client/scripts/views/pages/validators/substrate/identity.ts

```diff
@@ -1,31 +1,104 @@
 import m from 'mithril';
+import app from 'state';
+import { Button, Classes, Icons, Icon } from 'construct-ui';
+import User from 'views/components/widgets/user';
+import { ChainBase } from 'models';
+import { truncate } from 'lodash';
+import { externalLink } from 'helpers';
+import Substrate from 'controllers/chain/substrate/main';
+import { makeDynamicComponent } from 'models/mithril';
 import { IAccountInfo } from 'controllers/chain/substrate/staking';
 
-const Identity: m.Component<IAccountInfo, {}> = {
+const truncLength = {
+  length: 20
+};
+
+export interface IValidatorState {
+  dynamic: {
+    info: IAccountInfo;
+  },
+  isNominating: boolean;
+}
+
+export interface IdentityAttrs {
+  stash: string;
+}
+
+const Identity = makeDynamicComponent<IdentityAttrs, IValidatorState>({
+  getObservables: (attrs) => ({
+    // we need a group key to satisfy the dynamic object constraints, so here we use the chain class
+    groupKey: app.chain.class.toString(),
+    info: (app.chain.base === ChainBase.Substrate)
+      ? (app.chain as Substrate).staking.info(attrs.stash)
+      : null
+  }),
   view: (vnode) => {
+    const { info } = vnode.state.dynamic;
+    if (!info)
+      return m('span', [
+        m(User, { user: app.chain.accounts.get(vnode.attrs.stash), linkify: true }),
```

Comment:
Is this safe if we haven't loaded `accounts` yet?

Reply:
That's the pattern used all over. I am not sure, should implement a check before `accounts`?

Reply:
Usually we have some sort of check, like `app.chain?.loaded` -- I'm not sure if it's necessarily needed in this case, but wanted to make you aware that this call can fail if `app.chain` isn't loaded.